### DISCUSSION TITLE
JWT Asymmetrical keys

### DISCUSSION THREAD

**Discussion by JoeyyT** [nhost/hasura-backend-plus#644] [Q&A]
Hi,

I am looking to implement HBP in our stack of our software development company. One of our priorities is security and as I can see from the documentation the current method of JWT signing is through a secretKey. 

Is it currently possible to provide a private/public key combination to introduce the capability of other microservices to make use of only the public key instead of the global secret?

If this is not possible, would it be possible for me to create a PR and merge within a short time frame?

Many thanks!

**Comment by elitan** [comment]
Hey @JoeyyT 

We're currently not adding features to HBP and JWK was removed because we wanted to keep it simple.

If you want to give it a go with JWK that would be OK for us.

Internally we've frozen development on HBP and are focusing all our effort on a new auth service for Hasura. We hope to make it public in the coming weeks and we plan to add JWT (private/public) support for this new service.

**Reply by JoeyyT** [reply]
Hey @elitan ,

Sad to hear that development is frozen on HBP, however, good for us is that our primary focus with HBP is authentication and we are looking forward to the public announcement of this new service.

Can you disclose whether the public announcement includes a (stable) release, or is it just an update regarding the status of the development?

**Reply by elitan** [reply]
The announcement is the actual release and we'll tag it as a beta at first.

**Reply by JoeyyT** [reply]
@elitan Any information regarding the release? :)

**Comment by plmercereau** [comment]
RSA is not working anymore?
